.github/workflows/release.yaml

```diff
@@ -1,12 +1,46 @@
-name: Build Go release artifacts
+name: Semantic release
 
 on:
-  release:
-    types:
-      - created
+  push:
+    branches:
+      - main
+      - beta
+  workflow_dispatch:
 
 jobs:
+  release:
+    name: Semantic release
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: "☁️ checkout repository"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      - name: "🔧 setup node"
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: "🔧 install npm@latest"
+        run: npm i -g npm@latest
+
+      - name: "🚀 release"
+        id: semantic-release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NPM_PACKAGE_ROOT: "npm"
+        uses: open-sauced/release@v2.2.0
+
+    outputs:
+        release-tag: ${{ steps.semantic-release.outputs.release-tag }}
+
+        
   build:
+    needs:
+      - release
```

Comment:
Why does this need the `open-sauced/release` job to finish first? Can't they run in parallel since one is simply publishing the package to NPM and the other is actually building the binaries?

Reply:
My guess is since the release action will create GitHub tags and releases for us?

Reply:
https://github.com/open-sauced/pizza-cli/blob/73291d13d632b709f2583d834aefe6ad758de8d7/.github/workflows/release.yaml#L66
For the binaries to be uploaded, the release tag-name is needed which is available after the completion of the release job.